src/Server/WishList.fs

```diff
@@ -7,13 +7,16 @@ open Giraffe
 open ServerCode.Domain
 open ServerTypes
 open FSharp.Control.Tasks.V2
+open Thoth.Json.Net
 
 /// Handle the GET on /api/wishlist
 let getWishList (getWishListFromDB : string -> Task<WishList>) (token : UserRights) : HttpHandler =
      fun (next : HttpFunc) (ctx : HttpContext) ->
         task {
             let! wishList = getWishListFromDB token.UserName
-            return! ctx.WriteJsonAsync wishList
+            let json = WishList.Encoder wishList
+                        |> Encode.toString 0
+            return! ctx.WriteStringAsync json
```

Comment:
this doesn't set the HTTP headers correctly, right!?

Reply:
Probably no, I don't know Giraffe.

I guess adding an extension method like 

```fs
ctx.WriteJsonAsync(jsonValue : Encode.Value)
``` 

could be something to do in the future for SAFE apps.

You could also pass it the `value` and the `encoder` to apply:

```fs
ctx.WriteJsonAsync(value : 'T, Encode.Encoder<'T>)
```

I find the second option easier to read because people don't need to apply the `encoder` themself.

Reply:
See [this version](https://github.com/MangelMaxime/Thoth/issues/80#issuecomment-441707345). This sets the content type header and the status code. And should be more performant as well :)

Reply:
See [this version](https://github.com/MangelMaxime/Thoth/issues/80#issuecomment-441707345). This sets the content-type header and the status code. And should be also more performant :)

Reply:
We should just be using the `json` helper in Giraffe, having plugged in the appropriate Thoth helper at the start. The SAFE Template does this so there's a pattern that can be followed I think.

Reply:
@isaacabraham The problem, [as noted here](https://github.com/MangelMaxime/Thoth/issues/80#issuecomment-441591039), is the Giraffe.ThothSerializer only works with auto encoders. It can conflict with manual coders if they're used.

Reply:
Doh, sorry, I missed that.